src/golang/lib/execution_environment/execution_environment.go

```diff
@@ -61,8 +61,10 @@ func (e *ExecutionEnvironment) DeleteDBRecord(
 // Hash generates a hash based on the environment's
 // dependency set and python version.
 func (e *ExecutionEnvironment) Hash() (uuid.UUID, error) {
-	sliceToHash := append(e.Dependencies, e.PythonVersion)
-	sort.Strings(sliceToHash)
+	sliceToHash := make([]string, 0, len(e.Dependencies)+1)
```

Comment:
I had to modify this to a copy since `append` appear to modify `Dependencies` in-place and is injecting garbage to this field.